### PR TITLE
Purgedups

### PR DESCRIPTION
**Pull request by yumisims**
## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Review by muffato — CHANGES_REQUESTED** [review]
Also, the name of the module is meant to reflect the software package and tool name, not the pipeline. Here, the module is only for the FastK command of the FastK package, so the directory should be `modules/fastk/fastk`, leaving room for other FastK commands such as `Fastmerge`

File: tests/modules/purgedups/fastk/main.nf [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+nextflow.enable.dsl = 2
+
+include { PURGEDUPS_FASTK } from '../../../../modules/purgedups/fastk/main.nf'
+
+workflow test_purgedups_fastk {
+    
+        data = [
+        [ id:'test'], // meta map
+        [
+           file('/lustre/scratch123/tol/teams/grit/yy5/nf_test_small/ilDiaRubi1.trimmedReads.fasta.gz', checkIfExists: true),
```
Comment (muffato) [inline tests/modules/purgedups/fastk/main.nf:12]:
```suggestion
           file('https://tolit.cog.sanger.ac.uk/test-data/Diarsia_rubi/working/ilDiaRubi1.hicanu.20220110/subset/ilDiaRubi1.trimmedReads.fasta.gz', checkIfExists: true),
```

**Review by muffato — COMMENTED** [review]
1. I've made a slight modification to the Dockerfile to record the commit IDs that were used (since FastK is not properly versioned): https://gitlab.internal.sanger.ac.uk/tol-it/software/docker-images/-/merge_requests/5/diffs?commit_id=965adf60a693d8953018d2967ac479244602b9b3
The commit IDs are recorded in `/usr/local/src/*.version`. This bumps up the container version here
2. We now have the S3 bucket to host the test data, cf the README and the config file at https://github.com/sanger-tol/nf-core-modules/tree/test_data/tests/config
I have added your data

File: modules/purgedups/fastk/meta.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      type: file
+      description: compressed fasta file
+      pattern: "*.fasta.gz"
+
+  - kmer:
+    type: value
+    description: the size/length of kmer
+
+  - redir:
+    type:
```
[the remark this answers is not in the document]
Comment (yumisims) [reply modules/purgedups/fastk/meta.yml:30]: deleted

File: modules/purgedups/fastk/meta.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - versions:
+      type: file
+      description: File containing software versions
+      pattern: "versions.yml"
+  - bam:
+      type: file
+      description: Sorted BAM/CRAM/SAM file
+      pattern: "*.{bam,cram,sam}"
```
[the remark this answers is not in the document]
Comment (yumisims) [reply modules/purgedups/fastk/meta.yml:45]: deleted

**Comment by gq1** [comment]
The latest changes being being merged to this branches now. You can delete another merge request.

**Comment by muffato** [comment]
Hi Yumi.
For review, it's easier if you can make 1 pull request per software, e.g. just fastk here, and do genescope/purgedups in other PRs.

By the way, there is an ongoing pull-request on nf-core for fastk: https://github.com/nf-core/modules/pull/1739/files
It uses a Docker image EBP/Sweden people have built somewhere. I can't find exactly where the Dockerfile is, but it seems to be part of the genomeassembler pipeline they're working on (which should mimic our own), cf https://github.com/NBISweden/Earth-Biogenome-Project-pilot/issues/24

**Comment by yumisims** [comment]
Hi, I have modified the modules accordingly. 
I am also aware the module name conflicts with the one you mentioned.  The two modules are very smilar but are used for different purposes. Could you please suggest a name for my module to avoid the conflict. Thanks

**Comment by mahesh-panchal** [comment]
> It uses a Docker image EBP/Sweden people have built somewhere. I can't find exactly where the Dockerfile is

The Dockerfile is currently on another branch as I'm working on something else: https://github.com/NBISweden/Earth-Biogenome-Project-pilot/blob/feature_asm_validation/dockerfiles/fastk_genescopefk_merquryfk.Dockerfile

**Comment by muffato** [comment]
@yumisims : as far as I can see, the upstream module from @mahesh-panchal does the same thing, so no need to develop one in ToL. Can you instead review it ?

**Comment by mahesh-panchal** [comment]
I've Histex, GeneScopeFK, and MerquryFK waiting in the wings too.

**Pull request closed without merging**